### PR TITLE
Reorganize the setup and build instructions.

### PR DESCRIPTION
**Pull request by sulekhark**
This PR reorganizes the setup and build instructions based on the experience of a newcomer to the Checked C project. Further, these instructions were successfully executed by the summer intern Shiwei Weng when he setup his development environment for the first time. 

The instructions for Visual Studio build remain unchanged for the present and will be updated in a later PR, if necessary.

### CODE REVIEW

File: clang/docs/checkedc/Setup-and-Build.md [hunk 1]
```diff
@@ -57,138 +62,223 @@ need to end with line feeds. Visual Studio preserves line endings for files, so
...
+
+3. The Checked C language tests live in a folder within `llvm/project`. Change
+to the  `src/llvm/projects/checkedc-wrapper` directory and clone the Checked C
+repo:
+   ```
+   git clone https://github.com/Microsoft/checkedc
+   ```
+
+4. **\[OPTIONAL\]** Install `ccache` to speed up the compiler build on Linux and
+MacOS. [ ccache](https://ccache.samba.org) is a smart cache for GCC or Clang. It
```
Comment (mgrang) [inline clang/docs/checkedc/Setup-and-Build.md:135]: On Linux, we can install `cache` as `sudo apt install ccache`.

File: clang/docs/checkedc/Setup-and-Build.md [hunk 2]
```diff
@@ -57,138 +62,223 @@ need to end with line feeds. Visual Studio preserves line endings for files, so
...
+
+You will need to choose a drive that has at least 50 Gbytes free. You may need
+lots of space for the sources and the build. You can store the sources in any
+directory that you want. You should avoid spaces in parent directory names
+because this can confuse some tools.
+
+### Instructions for Linux:
+
+1. Choose any directory as your working directory. We will refer to to this
+directory as \<WORK_DIR\>.
```
Comment (mgrang) [inline clang/docs/checkedc/Setup-and-Build.md:115]: Linux uses forward slashes. I guess we can leave out the slashes around `WORK_DIR`.

File: clang/docs/checkedc/Setup-and-Build.md [hunk 3]
```diff
@@ -57,138 +62,223 @@ need to end with line feeds. Visual Studio preserves line endings for files, so
...
+6. Execute the following `cmake` command in the build directory:
+
+   ```
+   cmake -G Ninja -DLLVM_ENABLE_PROJECTS=clang   // Required to enable Clang build
+   -DCMAKE_INSTALL_PREFIX=<WORK_DIR>/install     // Directory where the compiler will be
+                                                 // installed when "ninja install" is executed. 
+   -DCMAKE_BUILD_TYPE=Release                    // Alternate value: Debug
+   -DLLVM_ENABLE_ASSERTIONS=ON                   // Alternate value: OFF
+   -DLLVM_CCACHE_BUILD=ON                        // OPTIONAL. If this definition exists, ccache
+                                                 // will be used to speen up builds.
```
Comment (mgrang) [inline clang/docs/checkedc/Setup-and-Build.md:162]: Typo: speen --> speed

File: clang/docs/checkedc/Setup-and-Build.md [hunk 4]
```diff
@@ -57,138 +62,223 @@ need to end with line feeds. Visual Studio preserves line endings for files, so
...
+   cd <WORK_DIR>/build
+   ```
+
+6. Execute the following `cmake` command in the build directory:
+
+   ```
+   cmake -G Ninja -DLLVM_ENABLE_PROJECTS=clang   // Required to enable Clang build
+   -DCMAKE_INSTALL_PREFIX=<WORK_DIR>/install     // Directory where the compiler will be
+                                                 // installed when "ninja install" is executed. 
+   -DCMAKE_BUILD_TYPE=Release                    // Alternate value: Debug
```
Comment (mgrang) [inline clang/docs/checkedc/Setup-and-Build.md:159]: Other supported values for `CMAKE_BUILD_TYPE` are `RelWithDebInfo` and `MinSizeRel`

File: clang/docs/checkedc/Setup-and-Build.md [hunk 5]
```diff
@@ -57,138 +62,223 @@ need to end with line feeds. Visual Studio preserves line endings for files, so
...
+      cd %TOP%\build
+      set PATH="C:\GnuWin32\bin";%PATH%
+      @call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
+      ```
+
+6. Execute the following `cmake` command in the same command shell created above
+in the `<WORK_DIR>\build` directory:
+
+      ```
+      cmake -G Ninja -DLLVM_ENABLE_PROJECTS=clang   // Required to enable Clang build
```
Comment (mgrang) [inline clang/docs/checkedc/Setup-and-Build.md:238]: We can describe these CMake options once to avoid repetition and maintenance.

Comment (sulekhark) [reply clang/docs/checkedc/Setup-and-Build.md:241]: In my experience, it is extremely helpful for first-time users to be provided command lines that they can virtually copy-paste in their setup. First-timers are usually overwhelmed by the number of unknown/unclear things and need instructions that do not have numerous redirections. So, I think that this repetition is useful and worth the extra maintenance effort.

File: clang/docs/checkedc/Setup-and-Build.md (hunk 3 above)
Comment (sulekhark) [reply clang/docs/checkedc/Setup-and-Build.md]: Done.

**Review by mgrang — APPROVED** [review]
LGTM.

**Review by kkjeer — APPROVED** [review]
LGTM.

**Pull request merged**